### PR TITLE
Make the compositor be the first program to launch

### PR DESCRIPTION
**Pull request by tintou**
Allows to make sure that the greeter will be launched when the compositor is ready, also allow us to use the right API on wayland.

### CODE REVIEW

**Comment by danirabbit** [comment]
Yay thank you! I keep meaning to look into this so that's really appreciated 🩷

**Comment by tintou** [comment]
Anyone can feel free to take over this MR as I only have limited time currently 👍

**Comment by tintou** [comment]
It is actually working when removing the switch to Gtk.Application so yay!

File: compositor/WindowManager.vala [hunk 1]
```diff
@@ -85,12 +83,6 @@ namespace GreeterCompositor {
 
             disable_tiling_shortcuts ();
 
-            fade_in_screen.save_easing_state ();
-            fade_in_screen.set_easing_duration (1000);
-            fade_in_screen.set_easing_mode (Clutter.AnimationMode.EASE);
-            fade_in_screen.opacity = 0;
-            fade_in_screen.restore_easing_state ();
```
Comment (danirabbit) [inline compositor/WindowManager.vala:92]: Was there a reason for removing the fade in? This branch seems to still work with it

**Review by danirabbit — COMMENTED** [review]
Can confirm it does work!

For some reason my install has started not starting the panel in either the greeter or the main session :thinking: so I don't know what's the cause. But also I'm not getting the wallpaper on the greeter anymore either. I can't confirm if it's related to this branch unfortunately.

But, this does seem to solve a major issue where sometimes the screen would just be black or only partially faded in. So functionally this is a big fix. I'd be in favor of merging and figuring out those regressions separately if anyone else is experiencing them.

Just one inline comment :)

**Review by danirabbit — APPROVED** [review]
Reverted changes to fadein. If that's important for something please propose it again separately :)

**Pull request merged**